crates/rpc/rpc-eth-api/src/core.rs

```diff
@@ -16,21 +13,40 @@ use reth_rpc_types::{
     simulate::{SimBlock, SimulatedBlock},
     state::{EvmOverrides, StateOverride},
     AnyTransactionReceipt, BlockOverrides, Bundle, EIP1186AccountProofResponse, EthCallResponse,
-    FeeHistory, Header, Index, RichBlock, StateContext, SyncStatus, Transaction,
-    TransactionRequest, Work,
+    FeeHistory, Header, Index, StateContext, SyncStatus, TransactionRequest, Work,
 };
 use tracing::trace;
 
+use crate::{
+    helpers::{
+        transaction::UpdateRawTxForwarder, EthApiSpec, EthBlocks, EthCall, EthFees, EthState,
+        EthTransactions, FullEthApi,
+    },
+    Block, Transaction,
+};
+
 /// Helper trait, unifies functionality that must be supported to implement all RPC methods for
 /// server.
-pub trait FullEthApiServer: EthApiServer + FullEthApi + UpdateRawTxForwarder + Clone {}
+pub trait FullEthApiServer:
+    EthApiServer<Transaction<Self::NetworkTypes>, Block<Self::NetworkTypes>>
+    + FullEthApi
+    + UpdateRawTxForwarder
+    + Clone
+{
+}
 
-impl<T> FullEthApiServer for T where T: EthApiServer + FullEthApi + UpdateRawTxForwarder + Clone {}
+impl<T> FullEthApiServer for T where
+    T: EthApiServer<Transaction<T::NetworkTypes>, Block<T::NetworkTypes>>
+        + FullEthApi
+        + UpdateRawTxForwarder
+        + Clone
+{
+}
 
 /// Eth rpc interface: <https://ethereum.github.io/execution-apis/api-documentation/>
 #[cfg_attr(not(feature = "client"), rpc(server, namespace = "eth"))]
 #[cfg_attr(feature = "client", rpc(server, client, namespace = "eth"))]
-pub trait EthApi {
+pub trait EthApi<T: RpcObject, B: RpcObject> {
```

Comment:
hmm, could we make this generic over EthApiTypes or Network?

not a blocker, but imo this would be nicer

Reply:
rpc derive macro doesn't allow it

Reply:
generic must impl `RpcObject`